src/writer.jl

```diff
@@ -0,0 +1,104 @@
+#
+# Writing Julia dictionaries to YAML files is implemented with multiple dispatch and recursion:
+# Depending on the value type, Julia chooses the appropriate _print function, which may add a
+# level of recursion.
+#
+
+"""
+    write_file(path, data, prefix="")
+
+Write some data (e.g. a dictionary or an array) to a YAML file.
+"""
+function write_file(path::AbstractString, data::Any, prefix::AbstractString="")
+    if (!endswith(path, ".yml"))
+        warn("The path $path does not end on '.yml'.")
+    end # suggest to use the appropriate file name extension, but do not throw an error
```

Comment:
here is too restrict on checking only ".yml, also the `warn` function is not defined I think in Julia 1.0+.
when I use path such as "./somepath.yaml", it triggered the warn function which is not defined in Julia 1.2.

Reply:
Thanks for the feedback! I removed the check to make it the user's responsibility to chose the file name that's most appropriate.

Reply:
great

Reply:
Could check here for `if !endswith(path, ".yml") && !endswith(path, ".yaml")` instead. Use `@warn "the path..."` instead.

But not really necessary I supposed